### PR TITLE
FORMS-7889 Number Input field accepting decimal value in runtime when…

### PR DESCRIPTION
**Pull request by dmaurya929**
… input type is selected as Integer.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes and the overall coverage did not decrease.
- [ ] All unit tests pass on CircleCi.
- [ ] I ran all tests locally and they pass.

### CODE REVIEW

**Comment by dmaurya929** [comment]
New PR of https://github.com/adobe/aem-core-forms-components/pull/637

File: ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/numberinput/v1/numberinput/clientlibs/site/js/numberinputwidget.js [hunk 1]
```diff
@@ -59,9 +57,9 @@ class NumericInputWidget {
         this.#model = model;
         // initialize options for backward compatibility
         this.#options = Object.assign({}, this.#defaultOptions, this.#model._jsonModel);
-        let matchStr =  this.#matchArray[this.#options.dataType];
+        var matchStr =  this.#matchArray[this.#options.type];
```
Comment (rismehta) [inline ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/numberinput/v1/numberinput/clientlibs/site/js/numberinputwidget.js:60]: Why did you change it to var from let ?

Comment (dmaurya929) [reply ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/numberinput/v1/numberinput/clientlibs/site/js/numberinputwidget.js:60]: We are redeclaring this `matchStr` variable in line 68, that's why `let` was throwing some error. I'm not sure how it was working earlier that's why just changed `var`

Comment (rismehta) [reply ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/numberinput/v1/numberinput/clientlibs/site/js/numberinputwidget.js:60]: `var` is not recommended anymore, even let should work for your use-case

**Review by rismehta — CHANGES_REQUESTED** [review]
-1 for now

File: ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/numberinput/v1/numberinput/clientlibs/site/js/numberinputwidget.js [hunk 2]
```diff
@@ -38,9 +38,7 @@
          }
          //TODO: to support writing in different locales \d should be replaced by [0-9] for different locales
          #matchArray ={
-             "integer":"^[+-]?{digits}*$",
-             "decimal":"^[+-]?{digits}{leading}({decimal}{digits}{fraction})?$",
-             "float":"^[+-]?{digits}*({decimal}{digits}*)?$"
+             "integer":"^[+-]?{digits}*$"
```
Comment (rismehta) [inline ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/numberinput/v1/numberinput/clientlibs/site/js/numberinputwidget.js:41]: Why have you removed this check for decimal and float ?

Comment (dmaurya929) [reply ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/numberinput/v1/numberinput/clientlibs/site/js/numberinputwidget.js:41]: In v2 we have 2 number type `Decimal` and `Integer`. For v2 we are using HTML`type='number'`, it is already equivalent to 'decimal`, we just need to validate for `integer` . 
In V1 we have `type='text'`, may be that's why we have those checks.
